src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -96,22 +99,39 @@ public StudyService(
         this.webClient =  webClientBuilder.build();
 
         this.studyRepository = studyRepository;
+        this.studyBySubjectRepository = studyBySubjectRepository;
+        this.studyByAccessRightsRepository = studyByAccessRightsRepository;
     }
 
-    Flux<StudyInfos> getStudyList() {
-        Flux<Study> studyList = studyRepository.findAll();
-        return studyList.map(study -> new StudyInfos(study.getName(), study.getDescription(), study.getCaseFormat()));
+    Flux<StudyInfos> getStudyList(String subject) {
+        Flux<StudyBySubject> studyBySubjectFlux = studyBySubjectRepository.findAllBySubject(subject);
+
+        return Flux.concat(studyBySubjectFlux.flatMap(studyBySubject -> {
+            Mono<Study> studyMono = studyRepository.findByName(studyBySubject.getStudyName());
+            return studyMono.map(study -> new StudyInfos(study.getName(), study.getDescription(), study.getCaseFormat()));
+        }), getPublicStudyList());
+    }
+
+    Flux<StudyInfos> getPublicStudyList() {
+        Flux<StudyByAccessRights> studyByAccessRightsFlux = studyByAccessRightsRepository.findAllByIsPrivate(false);
+        return studyByAccessRightsFlux.flatMap(studyByAccessRights -> {
+            Mono<Study> studyMono = studyRepository.findByName(studyByAccessRights.getStudyName());
```

Comment:
same remark, denormalize ?

Reply:
also keep only the public studies not from this user (because they are already included in the other request). Or alternatively, make the other request return only the private studies of the subject.